### PR TITLE
OPS-1555_Packages_and_Workflows

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

- Introduced the Package system: for sending groups of items around for workflow needs; the initial implementation supports `BliPackages`
- Updated the `/agreements/review/{id}` to reflect this new package; instead of immediately updating the status of the BLI, it will now create a package; which is sent to review.
- Introduced the `Workflow` system; which allows us to create generic workflow templates, and individual workflow instances; 
  - each workflow contains a set of steps; those steps are the steps/tasks that the workflow requires
  - Example: An agreement needs two BLI sent for approval, A workflow instance is created, with a single `WorkflowStepInstance` that will require a anyone in a pre-defined set of people to review, then approve.
  - THIS IS ONLY THE BUILDING BLOCK FOR WORKFLOWS - NOT YET PUT TO USE.

## Issue

#1555

## How to test

1. Navigate to the Agreement page.
2. Select an agreement with DRAFT BLIs and click the "send for review" button (plane button?)
3. Select the `Change Draft Budget Lines` Action
4. Select the BLIs to test
5. Add a note
6. Click Send to Approval

Expected Results:
You should still see an alert that the BLIs were sent for review, and be sent back to the `/agreements` page.
There should now be a new record in the `BliPackages` and `Packages` tables

## Definition of Done Checklist
- [ ] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

## Links

_If relevant, e.g. for a link to a piece of markdown documentation_

### CODE REVIEW

File: frontend/src/api/postBliPackage.js
Comment (stevtek) [inline frontend/src/api/postBliPackage.js:1]: I am wondering why this isn't a part of opsAPI.

Comment (tdonaworth) [reply frontend/src/api/postBliPackage.js:1]: It actually is - endpoint helpers were created in both formats

**Review by fpigeonjr — APPROVED** [review]
ran locally and was able to submit Agreement 1 for review.

File: frontend/Dockerfile.yarn [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM node:20-slim
```
[the remark this answers is not in the document]
Comment (tdonaworth) [reply frontend/Dockerfile.yarn:1]: This is `Dickerfile.yarn` - was only used during early yarn continuation - will delete.

File: frontend/src/pages/agreements/review/ReviewAgreement.jsx [hunk 2]
```diff
@@ -42,8 +44,10 @@ export const ReviewAgreement = () => {
     } = useGetAgreementByIdQuery(agreementId, {
         refetchOnMountOrArgChange: true
     });
+    const activeUser = useSelector((state) => state.auth.activeUser);
 
-    const [updateBudgetLineItemStatus] = useUpdateBudgetLineItemStatusMutation();
+    //const [updateBudgetLineItemStatus] = useUpdateBudgetLineItemStatusMutation();
```
[the remark this answers is not in the document]
Comment (tdonaworth) [reply frontend/src/pages/agreements/review/ReviewAgreement.jsx:49]: 👍

File: backend/data_tools/data/workflow_data.json5 [hunk 3]
```diff
@@ -0,0 +1,113 @@
+{
+  // workflow_type: [
```
[the remark this answers is not in the document]
Comment (tdonaworth) [reply backend/data_tools/data/workflow_data.json5:2]: keeping this as WIP for next-steps

**Pull request merged**